extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -22,19 +22,49 @@ import { AcctStore } from '../../../../js/common/platform/store/acct-store.js';
 import { FcUuidAuth } from '../../../../js/common/api/account-servers/flowcrypt-com-api.js';
 import { SmimeKey } from '../../../../js/common/core/crypto/smime/smime-key.js';
 import { PgpHash } from '../../../../js/common/core/crypto/pgp/pgp-hash.js';
+import { UploadedMessageData } from '../../../../js/common/api/account-server.js';
+import { ParsedKeyInfo } from '../../../../js/common/core/crypto/key-store-util.js';
+import { MultipleMessages } from './general-mail-formatter.js';
 
 export class EncryptedMsgMailFormatter extends BaseMailFormatter {
 
-  public sendableMsg = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingPrv?: Key): Promise<SendableMsg> => {
+  public sendableMsgs = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingKey?: ParsedKeyInfo): Promise<MultipleMessages> => {
     if (newMsg.pwd && !this.isDraft) {
       // password-protected message, temporarily uploaded (already encrypted) to:
       //    - flowcrypt.com/api (consumers and customers without on-prem setup), or
       //    - FlowCrypt Enterprise Server (enterprise customers with on-prem setup)
       //    It will be served to recipient through web
-      const { url: msgUrl, externalId } = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
+      const uploadedMessageData = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
       newMsg.pwd = undefined;
-      return await this.sendablePwdMsg(newMsg, pubkeys, { msgUrl, externalId }, signingPrv); // encrypted for pubkeys only, pwd ignored
-    } else if (this.richtext) { // rich text: PGP/MIME - https://tools.ietf.org/html/rfc3156#section-4
+      const collectedAttachments = await this.view.attachmentsModule.attachment.collectEncryptAttachments(pubkeys);
+      // we always have at least one pubkey, this is the main message, encrypted for pubkeys
```

Comment:
@tomholub Do I get it right? If all the recipients are pwd recipients, I should send pubkey-encrypted copy to myself? This is to be able to see the message content in sent messages and to not lose attachments (as attachments are still not uploaded to FES)?

Reply:
ping @tomholub

Reply:
Sorry - the more complicated the issue, the slower I'm to respond :-( I'll take a look

Reply:
> Do I get it right? If all the recipients are pwd recipients, I should send pubkey-encrypted copy to myself? 

It does sound odd. Let's not send a message to ourselves if there are no pubkey recipients.

A better fix should be done in the future. Maybe we will allow reading of FES messages from browser extension - whenever the browser extension notices a FES link in the message, it could load the message from it and render it.

> This is to be able to see the message content in sent messages

We'll have to fix that by loading and rendering FES messages, later

> and to not lose attachments (as attachments are still not uploaded to FES)?

Attachments are definitely uploaded to FES. That's why it takes so long to upload and that's why we cannot make 1 request to FES per FES recipient. We would end up uploading the attachments to FES 5 times. That's why we had to edit FES API to define several recipients in one request.

The attachments are encoded into a pgp/mime message and uploaded as a whole. It may appear there are no attachments in the code after it was encoded as pgp/mime, but try it out (on master): add attachments and you'll notice you can download them from FES. Or not?

Reply:
You're right. The attachments are indeed there.
What seems strange, why we have this <div> tag in the `text/plain` section, is this a bug? This also leads to an artifact when replying....
![image](https://user-images.githubusercontent.com/23335944/168438069-9b19794d-55b5-4d06-b2fe-53f38bcf2a51.png)

Reply:
The cryptup_reply is there on purpose, because FES and also legacy backend need it.

We will move away from it, and use mime headers instead, which I should have done from the beginning. Tech debt.

You can use `MsgBlockParser.stripFcTeplyToken` (lol - supposed to be called `MsgBlockParser.stripFcReplyToken`) to strip it away when you don't want it in the result.

Reply:
My question was actually: isn't it enough to have this `div` in `text/html` part of the multipart MIME message and strip it completely in `text/plain`? However, if you're going to scrap it anyway, no need to spend time on this.

Reply:
I guess you may be right, I'm not entirely sure (would have to be tested, and verified reply still works on both FES and flowcrypt.com/api). But as you say, it will get scrapped anyway.